anchore_engine/analyzers/hints.py

```diff
@@ -0,0 +1,406 @@
+import json
+
+from anchore_engine.subsys import logger
+from anchore_engine.utils import ensure_str
+
+
+class HintsError(Exception):
+    pass
+
+
+class BaseHint:
+    def __init__(self, pkg, type):
+        self.type = type
+        self.name = ensure_str(pkg.get("name", ""))
+        self.version = ensure_str(pkg.get("version", ""))
+        self.origin = ensure_str(pkg.get("origin", ""))
+
+    def check_required_fields(self):
+        if not all([self.name, self.version, self.type]):
+            raise HintsError(
+                "bad hints record, all hints records must supply at least a name, version and type"
+            )
+
+    def normalize(self):
+        pass
+
+    @staticmethod
+    def get_list_value(pkg, key):
+        if not key:
+            logger.debug("cannot get list value when key is not specified")
+            return []
+        value = ensure_str(pkg.get(key, ""))
+
+        # For Gem/NPM hints, we search both the singular and plural key name, where the plural is a list
+        list_value = pkg.get(key + "s", [])
+        if value and not list_value:
+            list_value = [value]
+        return list_value
+
+    def to_dict(self) -> dict:
+        self.check_required_fields()
+        self.normalize()
+        return {
+            "type": self.type,
+            "name": self.name,
+            "version": self.version,
+            "origin": self.origin,
+        }
+
+
+class RPMHint(BaseHint):
+    def __init__(self, pkg):
+        super().__init__(pkg, "rpm")
+        self.license = ensure_str(pkg.get("license", ""))
+        self.arch = ensure_str(pkg.get("arch", "x86_64"))
+        self.release = ensure_str(pkg.get("release", ""))
+        self.source = ensure_str(pkg.get("source", ""))
+        self.size = ensure_str(str(pkg.get("size", "0")))
+
+    def resolve_rpm_fields(self):
+        from anchore_engine.util.rpm import split_rpm_filename
+
+        p_name, p_parsed_version, p_release, p_epoch, p_arch = split_rpm_filename(
```

Comment:
What does p stand for here? Could it be spelled out?

Reply:
I think it stood for "parsed", a relic of the old hints code. Renamed!

Reply:
Thanks!